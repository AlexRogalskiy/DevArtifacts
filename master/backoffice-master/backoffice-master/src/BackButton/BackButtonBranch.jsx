import React from 'react'
import PropTypes from 'prop-types'

import {
  Button,
  withStyles,
} from '@material-ui/core'
import BackIcon from '@material-ui/icons/ArrowBackIos'

const styles = theme => ({
  leftIcon: {
    marginRight: theme.spacing.unit,
  },
})

const BackButtonBranch = ({
  onNavigateBack,
  classes,
}) => (
  <Button onClick={onNavigateBack}>
    <BackIcon className={classes.leftIcon} />
    Back
  </Button>
)

BackButtonBranch.propTypes = {
  onNavigateBack: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(BackButtonBranch)
