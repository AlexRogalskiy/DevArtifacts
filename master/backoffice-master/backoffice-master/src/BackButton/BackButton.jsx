import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'

import BackButtonBranch from './BackButtonBranch'

const withBackButton = Component => class extends React.Component {
  static propTypes = {
    url: PropTypes.string.isRequired,
    history: PropTypes.objectOf(PropTypes.any).isRequired,
  }

  constructor() {
    super()

    this.handleBack = this.handleBack.bind(this)
  }

  handleBack() {
    const { url, history } = this.props

    history.push(url)
  }

  render() {
    return (
      <Component
        {...this.props}
        onNavigateBack={this.handleBack}
      />
    )
  }
}

const BackButton = withBackButton(BackButtonBranch)

export default withRouter(BackButton)
