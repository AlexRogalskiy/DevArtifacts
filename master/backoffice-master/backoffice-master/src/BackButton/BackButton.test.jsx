import React from 'react'
import MockRouter from 'react-mock-router'
import { mount } from 'enzyme'

import BackButton from './'

describe('Back Button', () => {
  it('renders correctly', () => {
    const tree = mount((<MockRouter><BackButton url="/" /></MockRouter>))

    expect(tree).toMatchSnapshot()
  })

  it('updates history on click', () => {
    const push = jest.fn()
    const button = mount(<MockRouter push={push}><BackButton url="/" /></MockRouter>)

    button.find(BackButton).simulate('click')

    expect(push).toHaveBeenLastCalledWith('/')
  })
})
