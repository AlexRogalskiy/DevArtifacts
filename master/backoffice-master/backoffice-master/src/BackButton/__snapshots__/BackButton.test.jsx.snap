// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Button renders correctly 1`] = `
<button
  className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-31 BackButtonBranch-leftIcon-1"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    Back
  </span>
  <span
    className="MuiTouchRipple-root-38"
  />
</button>
`;
